Fixes #7669 by updating doc of Enumerable.Sum as suggested

eriawan opened the pull request:
## Summary

This PR fixes #7669 by updating `Enumerable.Sum` docs page with suggested change from the original issue.

cc @eiriktsarpalis @layomia

opbld33 commented:
Docs Build status updates of commit _[588b209](https://github.com/eriawan/dotnet-api-docs/commits/588b209187170b0552a3a141402d06f24a24b6a8)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[xml/System.Linq/Enumerable.xml](https://github.com/eriawan/dotnet-api-docs/blob/update-Enumerable-Sum-doc/xml/System.Linq/Enumerable.xml) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable?view=net-7.0&branch=pr-en-us-8377) |

For more details, please refer to the [build report](https://buildapi.docs.microsoft.com/Output/PullRequest/500c749d-de78-3d4a-00e4-269d599bf9ec/202209111931553720-8377/BuildReport?accessString=73b0b200f08468f32acad76e386b520127c51e0dc5dd96df12957cf70ffc415e).

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/500c749d-de78-3d4a-00e4-269d599bf9ec?tabName=brokenlinks).

For any questions, please:<ul><li>Try searching the docs.microsoft.com <a href="https://review.docs.microsoft.com/en-us/help/?branch=main">contributor guides</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Docs support channel</a></li></ul>

eiriktsarpalis commented on xml/System.Linq/Enumerable.xml:
```diff
@@ -11275,7 +11275,7 @@ If `count` is not a positive number, this method returns an identical copy of th
  This method returns zero if `source` contains no elements.
 
- The result does not include values that are `null`.
+ Items in `source` that are null are excluded from the computation of the sum. This method never returns `null`.
```
I would probably combine this with the sentence above to say something like:

> Items in `source` that are `null` are excluded from the computation of the sum. This method returns zero if `source` contains no elements or all elements are `null`.

eriawan replied:
@eiriktsarpalis 
Thanks for the feedback!
Done update the wording based on feedback above. Please review 🙂

opbld31 commented:
Docs Build status updates of commit _[b6a9e3e](https://github.com/eriawan/dotnet-api-docs/commits/b6a9e3e064fa75d2455649ad94db8e37e83f0bdf)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[xml/System.Linq/Enumerable.xml](https://github.com/eriawan/dotnet-api-docs/blob/update-Enumerable-Sum-doc/xml/System.Linq/Enumerable.xml) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable?view=net-7.0&branch=pr-en-us-8377) |

For more details, please refer to the [build report](https://buildapi.docs.microsoft.com/Output/PullRequest/500c749d-de78-3d4a-00e4-269d599bf9ec/202209142056487386-8377/BuildReport?accessString=45a086cd33c2149affbf8f282916b12fd7dbcceaf9467ff2fa85417cfad5aac4).

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/500c749d-de78-3d4a-00e4-269d599bf9ec?tabName=brokenlinks).

For any questions, please:<ul><li>Try searching the docs.microsoft.com <a href="https://review.docs.microsoft.com/en-us/help/?branch=main">contributor guides</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Docs support channel</a></li></ul>

opbld33 commented:
Docs Build status updates of commit _[305fe1a](https://github.com/eriawan/dotnet-api-docs/commits/305fe1aa67bb37b0d54429fc77f5ffdef515e336)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[xml/System.Linq/Enumerable.xml](https://github.com/eriawan/dotnet-api-docs/blob/update-Enumerable-Sum-doc/xml/System.Linq/Enumerable.xml) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable?view=net-7.0&branch=pr-en-us-8377) |

For more details, please refer to the [build report](https://buildapi.docs.microsoft.com/Output/PullRequest/500c749d-de78-3d4a-00e4-269d599bf9ec/202209142105154912-8377/BuildReport?accessString=ca7268f9869d07b649edf40bae38ed7abfde05f29161ec73e0d01f926f800d04).

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/500c749d-de78-3d4a-00e4-269d599bf9ec?tabName=brokenlinks).

For any questions, please:<ul><li>Try searching the docs.microsoft.com <a href="https://review.docs.microsoft.com/en-us/help/?branch=main">contributor guides</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Docs support channel</a></li></ul>

eiriktsarpalis commented on xml/System.Linq/Enumerable.xml:
```diff
@@ -11993,9 +11983,7 @@ If `count` is not a positive number, this method returns an identical copy of th
- This method returns zero if `source` contains no elements.
-
- The result does not include values that are `null`.
+ Items in `source` that are null are excluded from the computation of the sum. This method never returns `null`.
```
Any particular reason why this remark uses different wording than the ones above?

eiriktsarpalis replied:
Right, my point is that it's not applied to all methods consistently. Shouldn't it be the same everywhere?

eriawan replied:
I see, I understand now,  thanks! 
I was thinking that the wordings only applicable to Sum that takes nullable parameters such as `int?` 
Ok, I'm going to apply to all methods as well today. 🙂

eiriktsarpalis replied:
Are some of these overloads not taking nullable parameters? I haven't checked, but I would assume they do because the remarks already make reference to `null` values.

eriawan replied:
> Are some of these overloads not taking nullable parameters? I haven't checked, but I would assume they do because the remarks already make reference to `null` values.

AFAIK yes. For example:
https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.sum?view=net-6.0#system-linq-enumerable-sum(system-collections-generic-ienumerable((system-decimal)))

and looking at the remark, I think it doesn't have reference to `null` values:

![image](https://user-images.githubusercontent.com/8773147/190429819-e9cd4488-9120-416d-a542-feb548197496.png)

CMIIW, I think I need to ask your confirmation about this, because going back to my original understanding before, the update that I have applied should only matter for sum that takes IEnumerable<T> that takes T nullable such as `int?` or `double?`, like below overloads:

![image](https://user-images.githubusercontent.com/8773147/190431368-05afb23a-6972-4b18-9638-802aef61e44c.png)

eiriktsarpalis replied:
All remark sections that this PR updates originally contain this comment:

> The result does not include values that are `null`.

Presumably they're all nullable variants? If so then the same change should be applied everywhere.

eriawan replied:
> Presumably they're all nullable variants? If so then the same change should be applied everywhere.

Yes, I've only updated all of the remarks that have that sentence and as long as the IEnumerable parameter has with T of nullable primitive number types such as `int?`, `long?`, `Double?`.

Ok, let me check again and ensure that I'm going to update all of the remarks section of any Sum that has met the above requirements.

gewarren commented on xml/System.Linq/Enumerable.xml:
```diff
@@ -11273,9 +11273,7 @@ If `count` is not a positive number, this method returns an identical copy of th
- This method returns zero if `source` contains no elements.
-
- The result does not include values that are `null`.
+ Items in `source` that are `null` are excluded from the computation of the sum. This method returns zero if `source` contains no elements or all elements are `null`.
```
Since you are changing all these remarks, could you also add a blank line after the heading:

```suggestion

Items in `source` that are `null` are excluded from the computation of the sum. This method returns zero if `source` contains no elements or all elements are `null`.
```

eriawan replied:
updared

eriawan replied:
@gewarren 
done applying feedback on `Sum()`. Please review.

eiriktsarpalis commented:
@eriawan would it be possible to address the remaining feedback? Thanks!

eriawan commented:
@eiriktsarpalis 
Yes, I'm addressing this. Apologize, I've been busy for the last 10 days.
I'll update this Saturday.

opbld34 commented:
Learn Build status updates of commit _[cecd26a](https://github.com/eriawan/dotnet-api-docs/commits/cecd26a3b1b02f2b7df4af169494e0210b2af4d3)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[xml/System.Linq/Enumerable.xml](https://github.com/eriawan/dotnet-api-docs/blob/update-Enumerable-Sum-doc/xml/System.Linq/Enumerable.xml) | :white_check_mark:Succeeded | [View](https://review.learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable?view=net-7.0&branch=pr-en-us-8377) |

For more details, please refer to the [build report](https://buildapi.docs.microsoft.com/Output/PullRequest/500c749d-de78-3d4a-00e4-269d599bf9ec/202210081821183323-8377/BuildReport?accessString=497761e58504439a93c44b306e1364de28b308537f6ef53d9e32aeee1b8f1b82).

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/500c749d-de78-3d4a-00e4-269d599bf9ec?tabName=brokenlinks).

For any questions, please:<ul><li>Try searching the learn.microsoft.com <a href="https://review.learn.microsoft.com/en-us/help/?branch=main">contributor guides</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Learn support channel</a></li></ul>

eriawan commented:
I'm still going thru all of the usage of Nullable structs. Going to finish this today.

eriawan commented:
@eiriktsarpalis 

As far as I know I have addressed all feedback, including feedback from @gewarren , focusing on `Sum()` mehods only.

@gewarren 

There are some whitespaces on other methods of Enumerable. In this PR I'm focusing on `Sum()` method and its overloads first. then I'll update on separate PR to ensure that this PR review should be focusing on Sum() first, IMHO.

Thanks in advance! 🙂

opbld31 commented:
Learn Build status updates of commit _[174a90a](https://github.com/eriawan/dotnet-api-docs/commits/174a90a93d4e31ca187f067582a2ea1162ac8271)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[xml/System.Linq/Enumerable.xml](https://github.com/eriawan/dotnet-api-docs/blob/update-Enumerable-Sum-doc/xml/System.Linq/Enumerable.xml) | :white_check_mark:Succeeded | [View](https://review.learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable?view=net-7.0&branch=pr-en-us-8377) |

For more details, please refer to the [build report](https://buildapi.docs.microsoft.com/Output/PullRequest/500c749d-de78-3d4a-00e4-269d599bf9ec/202210092221416818-8377/BuildReport?accessString=3b9500e49e984c86e6c146d63c8bc823478aea22863109d0272616d23ddfaba9).

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/500c749d-de78-3d4a-00e4-269d599bf9ec?tabName=brokenlinks).

For any questions, please:<ul><li>Try searching the learn.microsoft.com <a href="https://review.learn.microsoft.com/en-us/help/?branch=main">contributor guides</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Learn support channel</a></li></ul>

eiriktsarpalis reviewed (approved):
Thanks!
